dependencies/src/test/java/com/google/cloud/tools/opensource/dependencies/DependencyGraphIntegrationTest.java

```diff
@@ -104,7 +104,7 @@ public void testJaxen()
     Truth.assertThat(updates).hasSize(5);
     
     List<DependencyPath> conflicts = graph.findConflicts();
-    Truth.assertThat(conflicts).hasSize(33);
+    Truth.assertThat(conflicts).hasSize(30);
```

Comment:
Pre-order and level-order generated different size of conflict. Noted the items in
https://github.com/GoogleCloudPlatform/cloud-opensource-java/issues/162

Reply:
it would be worth understanding why. What changed in the output?

Reply:
Checking...

Reply:
When it comes to a cyclic dependency, there's a difference in the returned graph of getCompleteDependencies between pre-order and level-order, because the two algorithm grow their dependency trees in different order. When they stop building dependency trees due to a cyclic dependency, the resulting (incomplete) trees look different.

`findConflicts` uses the tree to count the number of conflicting artifacts, the result differs between (incomplete) tree by pre-order and another tree by level-order.

https://github.com/GoogleCloudPlatform/cloud-opensource-java/issues/162#issuecomment-430063293

Among the dependency involved in this `testJaxen`, [dom4j:1.5.2](https://search.maven.org/artifact/dom4j/dom4j/1.5.2/jar) and [jaxen:1.1-beta-4](https://search.maven.org/artifact/jaxen/jaxen/1.1-beta-4/jar) have cyclic dependency.

Reply:
When it comes to a cyclic dependency, there's a difference in the returned graph of getCompleteDependencies between pre-order and level-order, because the two algorithm grow their dependency trees in different order. When they stop building dependency trees due to a cyclic dependency, the resulting (incomplete) trees look different.

`findConflicts` uses the tree to count the number of conflicting artifacts, the result differs between an incomplete tree by pre-order and another incomplete tree by level-order.

https://github.com/GoogleCloudPlatform/cloud-opensource-java/issues/162#issuecomment-430063293

Among the dependency involved in this `testJaxen`, [dom4j:1.5.2](https://search.maven.org/artifact/dom4j/dom4j/1.5.2/jar) and [jaxen:1.1-beta-4](https://search.maven.org/artifact/jaxen/jaxen/1.1-beta-4/jar) have cyclic dependency.